pkg/odo/cli/init/init.go

```diff
@@ -145,6 +146,14 @@ func (o *InitOptions) Run() (err error) {
 	if err != nil {
 		return err
 	}
+
+	// Set the name in the devfile but do not write it yet to disk,
+	// because the starter project downloaded at the end might come bundled with a specific Devfile.
+	name, err := o.clientset.InitClient.PersonalizeName(devfileObj, o.flags)
```

Comment:
Can you add an integration test to tests/integration/devfile/cmd_devfile_init_test.go as well to cover this scenario for the non-interactive mode?
`odo init --name aname --devfile-path tests/examples/source/devfiles/nodejs/devfile-with-starter-with-devfile.yaml` and ensure `aname` is set in the downloaded devfile.

Reply:
I assumed there would already exist test cases for such a scenario. But indeed, it looks like this is tested only via `odo create`. I will add a test case for this. Thanks.

Reply:
@valaparthvi Done - please take a look at [this commit](https://github.com/redhat-developer/odo/pull/5553/commits/97d282b3e7f8c9946859a88728dfb542cd6289d2)